docs/specs/lsp.yml

```diff
@@ -123,3 +123,63 @@ languageServer:
         "uri": "file:///{DOCSET_PATH}/b.md",
         "diagnostics": [{ "code": "file-not-found" }]
       }
+---
+# Config open/change (non-saved) should not trigger build
+os: windows
+inputs:
+  docfx.yml:
+  a.md:
+languageServer:
+  - notification: "textDocument/didOpen"
+    params:
+      {
+        "textDocument": {
+          "uri": "file:///{DOCSET_PATH}/a.md",
+          "text": "[Test](non-existed.md)"
+        }
+      }
+  - expectNotification: "textDocument/publishDiagnostics"
+    params:
+      {
+        "uri": "file:///{DOCSET_PATH}/a.md",
+        "diagnostics": [{ "code": "file-not-found" }]
+      }
+  - notification: "textDocument/didOpen"
+    params:
+      {
+        "textDocument": {
+          "uri": "file:///{DOCSET_PATH}/docfx.yml",
+          "text": "Invalid change"
+        }
+      }
+  - expectNoNotification: true
+---
+# Open/Change of files out of current repo should not trigger build
+os: windows
+inputs:
+  docfx.yml:
+  a.md:
+languageServer:
+  - notification: "textDocument/didOpen"
+    params:
+      {
+        "textDocument": {
+          "uri": "file:///{DOCSET_PATH}/a.md",
+          "text": "[Test](non-existed.md)"
+        }
+      }
+  - expectNotification: "textDocument/publishDiagnostics"
+    params:
+      {
+        "uri": "file:///{DOCSET_PATH}/a.md",
+        "diagnostics": [{ "code": "file-not-found" }]
+      }
+  - notification: "textDocument/didOpen"
+    params:
+      {
+        "textDocument": {
+          "uri": "file:///{DOCSET_PATH}/../a.md",
+          "text": "Invalid change"
+        }
+      }
+  - expectNoNotification: true
```

Comment:
Does it mean this line will wait for 20 seconds?

Reply:
yes